src/backend/distributed/transaction/transaction_management.c

```diff
@@ -320,6 +321,7 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			UnSetDistributedTransactionId();
 
 			PlacementMovedUsingLogicalReplicationInTX = false;
+			ShardMovedUsingLogicalReplicationInTX = false;
```

Comment:
shouldn't we reuse the same flag? We do not want move+split in the same transaction.

Reply:
It makes sense if we do not allow move+split in the same transaction.